### PR TITLE
ADS: Allow set endIcon programmatically on TextInput component

### PR DESCRIPTION
**Pull request by nalcalag**
Task/Issue URL: https://app.asana.com/0/488551667048375/1204667366884988/f

### Description
Added `fun setEndIcon(endIconRes: Int?, contentDescription: String? = null)` method to TextInput interface so we can edit it programmatically and not only from the xml file

### Steps to test this PR (Optional)

- Install from this branch
- Go to Settings > App Components Design Preview
- Select Text Input tab
- [ ] Check there is an end icon set on the 3rd item (Non-editable text with end icon)

### CODE REVIEW

**Comment by nalcalag** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #3216** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/3216" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/duckduckgo/Android/3216?utm_source=stack-comment).

File: common-ui/src/main/res/layout/component_text_input_view.xml [hunk 1]
```diff
@@ -58,8 +58,7 @@
             android:layout_marginTop="@dimen/keyline_3"
             android:hint="Non-editable text with end icon"
             android:text="This is not editable."
-            app:editable="false"
-            app:endIcon="@drawable/ic_copy" />
```
Comment (malmstein) [inline common-ui/src/main/res/layout/component_text_input_view.xml:62]: why is this removed?

File: common-ui/src/main/java/com/duckduckgo/mobile/android/ui/view/text/DaxTextInput.kt [hunk 2]
```diff
@@ -52,6 +54,10 @@ interface TextInput {
 
     fun addTextChangedListener(textWatcher: TextWatcher)
     fun removeTextChangedListener(textWatcher: TextWatcher)
+    fun setEndIcon(
+        @DrawableRes endIconRes: Int?,
```
Comment (malmstein) [inline common-ui/src/main/java/com/duckduckgo/mobile/android/ui/view/text/DaxTextInput.kt:58]: I don’t think we should allow this to be nullable

**Review by malmstein — APPROVED** [review]
Couple comments but nothing blocking, just the nullable piece that I think we should change

File: common-ui/src/main/java/com/duckduckgo/mobile/android/ui/view/text/DaxTextInput.kt (hunk 2 above)
Comment (nalcalag) [reply common-ui/src/main/java/com/duckduckgo/mobile/android/ui/view/text/DaxTextInput.kt:58]: it's to remove the icon. Another approach could be to create a different method to `hideEndIcon()` but I think this is simpler.

File: common-ui/src/main/res/layout/component_text_input_view.xml (hunk 1 above)
Comment (nalcalag) [reply common-ui/src/main/res/layout/component_text_input_view.xml:62]: Good catch! `endIcon` is set programmatically with the new method. But I will bring back `editable="false"`

Comment (nalcalag) [reply common-ui/src/main/res/layout/component_text_input_view.xml:62]: Actually, `editable="false"` is still there so the change it's just show endIcon programmatically so there is a use case in the preview which is intended.

File: common-ui/src/main/java/com/duckduckgo/mobile/android/ui/view/text/DaxTextInput.kt (hunk 2 above)
Comment (nalcalag) [reply common-ui/src/main/java/com/duckduckgo/mobile/android/ui/view/text/DaxTextInput.kt]: Talk to @malmstein and we agreed on add `removeEndIcon()` so `endIconRes` is not nullable anymore
